src/cameras/capture/opencv_camera/opencv_camera.py

```diff
@@ -108,15 +144,23 @@ def _create_thread(self):
             webcam_id=self.webcam_id_as_str,
             get_next_frame=self.get_next_frame,
             save_video=self._config.save_video,
-            frame_width=self.get_frame_width(),
-            frame_height=self.get_frame_height(),
+            frame_width=self.image_width,
+            frame_height=self.image_height,
         )
 
-    def get_frame_width(self):
-        return int(self._opencv_video_capture_object.get(3))
+    @property
+    def image_width(self):
+        try:
```

Comment:
if we try to grab the image width and it fails, theres an issue with the software and we should raise an exception from the class.

Reply:
I added a `OpenCVCameraError` exception to that file, but I've never worked with custom exceptions and I don't know how to evoke that error intentionally so I'm not totally certain it works


(I also don't know how to reference specific lines and/or commits here...)

Reply:
Custom exception creation isn't particularly necessary. If we were releasing an SDK, then I would worry about something like that, and that would be created nearer to the customer (a.k.a developer) API rather than in low level classes that we use to handle our shit :D

Mainly, if we try to call `.get` on `None` or an OpenCV object that isn't ready, an exception will naturally be thrown, and we want that. `image_width` should just perform the call to `.get` and leave try/except out.

Reply:
got it, thanks!